### ISSUE TITLE
HHVM CLI script hangs in vsdebug mode after uncaught exception, even with no debugger attached

### ISSUE THREAD

**Issue by enjmusic** [facebook/hhvm#8484]
### HHVM Version

```raw
$ hhvm --version
HipHop VM 3.30.3 (rel)
Compiler: remotes/origin/3.30.3-slack-0-g5a3df294bf9d921093eac96d93aff8f57b0779e9
Repo schema: fb5f31ec9856182de174e17a7439c92f221ad9a7
```

```raw
$ hh_client --version
hh--3.30.3 Mon Apr  1 17:25:31 2019
```

### Operating System and Version

Ubuntu 16.04

```raw
$ uname -a
Linux <hostname_redacted> 4.4.0-1075-aws #85-Ubuntu SMP Thu Jan 17 17:15:12 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

### Standalone code, or other way to reproduce the problem

`repro.php`:
```hack
<?hh // strict

<<__EntryPoint>>
function vsdebug_repro_main(): void {
    throw new Exception("Ahh!");
}
```

Run with:
```raw
hhvm --mode vsdebug --vsDebugPort 8998 --vsDebugNoWait true repro.php
```

### Actual result

```raw
$ hhvm --mode vsdebug --vsDebugPort 8998 --vsDebugNoWait true repro.php

Fatal error: Uncaught exception 'Exception' with message 'Ahh!' in <redacted>/repro.php:7
Stack trace:
#0 (): vsdebug_repro_main()
#1 {main}
![fatal] [Fri Apr 12 14:33:48 2019] Uncaught Exception: Ahh! in <redacted>/repro.php on line 7 (#vsdebug_repro_main())
^C
```
Hangs after the `![fatal]` line even if no debugger is attached.

### Expected result
The CLI script exits when encountering the fatal, at least if no debugger is attached.

**Comment by ebluestein** [comment]
@enjmusic  thanks for the report and the awesomely narrowed down repro case! I'm able to repro this and should have a fix in shortly.

**Comment by enjmusic** [comment]
@ebluestein Just a clarification - is the way it's working expected behavior for CLI mode? I still wasn't totally sure whether this felt unintentional or not.

**Comment by ebluestein** [comment]
No, the entire process should still exit cleanly when the script exits if run in CLI mode. There was a bug in the debugger shutdown path causing it to hang if no client was ever attached in this case.

**Comment by enjmusic** [comment]
@ebluestein Just checking in, do you have an idea of which release of HHVM this fix will be in?

**Comment by fredemmott** [comment]
Should be fixed by ed1731ce702769987d19fc5183d0c3c950f9deda , which was in 4.2.0
